src/main/java/com/celements/model/access/DefaultModelAccessFacade.java

```diff
@@ -633,6 +634,38 @@ private BaseProperty getBaseProperty(BaseObject obj, String key) {
   }
 
   @Override
+  public <T> Optional<T> getFieldValue(XWikiDocument doc, ClassField<T> field) {
+    return Optional.fromNullable(getProperty(checkNotNull(doc), checkNotNull(field)));
+  }
+
+  @Override
+  public <T> Optional<T> getFieldValue(DocumentReference docRef, ClassField<T> field)
+      throws DocumentNotExistsException {
+    return Optional.fromNullable(getProperty(checkNotNull(docRef), checkNotNull(field)));
+  }
+
+  @Override
+  public <T> Optional<T> getFieldValue(XWikiDocument doc, ClassField<T> field, T ignoreValue) {
+    checkNotNull(ignoreValue);
+    Optional<T> property = getFieldValue(doc, field);
+    if (property.isPresent() && Objects.equal(property.get(), ignoreValue)) {
+      property = Optional.absent();
+    }
+    return property;
+  }
+
+  @Override
+  public <T> Optional<T> getFieldValue(DocumentReference docRef, ClassField<T> field, T ignoreValue)
+      throws DocumentNotExistsException {
+    checkNotNull(ignoreValue);
+    Optional<T> property = getFieldValue(docRef, field);
+    if (property.isPresent() && Objects.equal(property.get(), ignoreValue)) {
+      property = Optional.absent();
+    }
+    return property;
+  }
+
+  @Override
   public <T> T getProperty(DocumentReference docRef, ClassField<T> field)
```

Comment:
perhaps we want to discourage using this method in favour of Optional return values by marking it as deprecated?